src/script/printer/relax/utils.h

```diff
@@ -82,10 +84,47 @@ inline Optional<ExprDoc> StructInfoAsAnn(const relax::Var& v, const ObjectPath&
   if (!v->struct_info_.defined()) {
     return NullOpt;
   }
+  bool attempt_to_hide_struct_info = !d->cfg->show_all_struct_info;
+
   if (const auto* call = rhs.as<relax::CallNode>()) {
     static const Op& call_tir_op = Op::Get("relax.call_tir");
     static const Op& call_dps_packed_op = Op::Get("relax.call_dps_packed");
     if (call->op.same_as(call_tir_op) || call->op.same_as(call_dps_packed_op)) {
+      attempt_to_hide_struct_info = true;
+    }
+  }
+  if (attempt_to_hide_struct_info) {
```

Comment:
Hm, I wonder if this is more complicated than what we would really want. Perhaps a syntactic rule for when to write StructInfo might make more sense? E.g., have it for op calls but not tuple indices, etc.? I'm not sure what situations are sufficiently non-obvious to require an annotation if a user wants to hide them. It's not wrong, but it seems a little unusual for the TVMScript printer to be running type inference itself.

Reply:
My worry with having separate rules for it is that those rules could diverge from the actual struct inference.  Even when the syntactic rules allow the struct info to be hidden, it would still need to compare against the actual struct info to see if the annotations provide different information.  That comparison would require StructInfo to compare against, and so this function would effectively be another implementation of `InferStructInfo`, but one which could get out of sync with the canonical version.

By implementing it in terms of the struct inference, the same inference rules are used for both parsing and printing.  Just as parsing can fill in missing information by calling the type inference, the printing can check if information can be omitted by calling the type inference.

Reply:
It seems a little like overkill, but we _are_ running the normalizer between every pass anyway so I guess it's not so bad.

Reply:
Yeah, probably overkill, but I tend to be more okay with overkill for non-default behavior.  Good point on the comparison with normalizing on each pass.